### PR TITLE
chore: refactor docker-compose files/options

### PR DESCRIPTION
**Pull request by obdulia-losantos**
Finishes: https://jira.ehealthafrica.org/browse/AET-100

Extracted common settings to `docker-compose-base.yml`. The rest of files extend and override the services declared there.

### CODE REVIEW

File: docker-compose-base.yml [hunk 1]
```diff
@@ -0,0 +1,188 @@
+# ------------------------------------------------------------------------------
+# Base settings for these containers:
+#
+#  * Databases + NGINX containers (use same settings in all files)
+#  * Aether Kernel
+#  * ODK Collect Adapter
```
Comment (admbtlr) [inline docker-compose-base.yml]: Let's refer to it as the `ODK Module` everywhere

File: docker-compose-base.yml [hunk 2]
```diff
@@ -0,0 +1,188 @@
...
+#  * Mobile CouchDB Sync
```
Comment (admbtlr) [inline docker-compose-base.yml]: Likewise, `CouchDB Sync Module` (it's not necessarily just for mobile)

File: docker-compose-base.yml [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+            - "8000:8000"
+        command: start_dev
+
+
+    # ---------------------------------
+    # ODK Adapter container
+    # ---------------------------------
+
+    odk-base:
+        build: aether-odk-importer
```
Comment (admbtlr) [inline docker-compose-base.yml]: No `importer` ever! It should be `aether-odk-module`

File: docker-compose-base.yml [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+            - "8443:8443"
+        command: start_dev
+
+
+    # ---------------------------------
+    # Mobile Sync containers
+    # ---------------------------------
+
+    sync-base:
+        build: aether-couchdb-sync
```
Comment (admbtlr) [inline docker-compose-base.yml]: `aether-couchdb-sync-module`

**Review by admbtlr — CHANGES_REQUESTED** [review]
I guess you're pretty sick of renaming things by now, so I can do this if you want... :)

File: docker-compose-base.yml (hunk 3 above)
Comment (obdulia-losantos) [reply docker-compose-base.yml]: Do you mean `odk-base` to `aether-odk-module`?

Comment (admbtlr) [reply docker-compose-base.yml]: I actually meant the `build` and `image` values

File: docker-compose-base.yml [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+            RDS_PASSWORD: ""
+            RDS_PORT: 5432
+            RDS_USERNAME: postgres
+
+            # media folder
+            MEDIA_ROOT: /tmp/
+
+            WEB_SERVER_PORT: 8443
+        volumes:
+            - ./aether-odk-importer:/code
```
Comment (admbtlr) [inline docker-compose-base.yml]: And here too...

File: docker-compose-base.yml (hunk 3 above)
Comment (obdulia-losantos) [reply docker-compose-base.yml]: `build` is the folder name... are you sure? No problem on that but need confirmation :writing_hand:

File: README.md [hunk 6]
```diff
@@ -82,22 +82,22 @@ of the most common ones with non default values. For more info take a look at th
 - `WEB_SERVER_PORT`: `8000` Web server port.
 
 
-#### Aether ODK Importer
+#### Aether ODK
```
Comment (admbtlr) [inline README.md]: **Aether ODK Module**

File: README.md [hunk 7]
```diff
@@ -82,22 +82,22 @@ of the most common ones with non default values. For more info take a look at th
...
+- `RDS_DB_NAME`: `odk` Postgres database name.
 - `WEB_SERVER_PORT`: `8443` Web server port.
 - `AETHER_KERNEL_TOKEN`: `a2d6bc20ad16ec8e715f2f42f54eb00cbbea2d24` Token to connect to kernel server.
 - `AETHER_KERNEL_URL`: `http://kernel:8000` Aether Kernel Server url.
-- `AETHER_KERNEL_URL_TEST`: `http://kernel-test:9000` Aether Kernel Server url.
+- `AETHER_KERNEL_URL_TEST`: `http://kernel-test:9000` Aether Kernel Testing Server url.
 
 
-#### Aether Couchdb Sync
+#### Aether CouchDB Sync
```
Comment (admbtlr) [inline README.md]: **Aether CouchDB Sync Module**

File: docker-compose-base.yml [hunk 8]
```diff
@@ -0,0 +1,183 @@
+# ------------------------------------------------------------------------------
+# Base settings for these containers:
+#
+#  * Databases + NGINX containers (use same settings in all files)
+#  * Aether Kernel Module
```
Comment (admbtlr) [inline docker-compose-base.yml]: `Aether Kernel` 

The kernel isn't a module - we have 1 kernel and n modules. Of the 3 modules that we currently have (ODK, CouchDB Sync and Gather), 2 are "core modules" (ODK and CouchDB Sync), while one is a "project module" (Gather).

File: docker-compose-base.yml [hunk 9]
```diff
@@ -0,0 +1,183 @@
...
+      # media folders per container
+      - ./tmp/kernel:/media/kernel
+      - ./tmp/odk:/media/odk
+      - ./tmp/sync:/media/sync
+    ports:
+      - "80:80"
+
+
+  # ---------------------------------
+  # Aether Kernel module
```
Comment (admbtlr) [inline docker-compose-base.yml]: `Aether Kernel`

File: docker-compose-base.yml [hunk 10]
```diff
@@ -0,0 +1,183 @@
...
+  # ---------------------------------
+
+  kernel-base:
+    build: aether-kernel-module
```
Comment (admbtlr) [inline docker-compose-base.yml]: `aether-kernel`

File: docker-compose-base.yml [hunk 11]
```diff
@@ -0,0 +1,183 @@
...
+    image: aether-kernel-module
```
Comment (admbtlr) [inline docker-compose-base.yml]: `aether-kernel`

File: docker-compose-base.yml [hunk 12]
```diff
@@ -0,0 +1,183 @@
...
+
+      RDS_DB_NAME: aether
+      RDS_HOSTNAME: db
+      RDS_PASSWORD: ""
+      RDS_PORT: 5432
+      RDS_USERNAME: postgres
+
+      WEB_SERVER_PORT: 8000
+    volumes:
+      - ./aether-kernel-module:/code
```
Comment (admbtlr) [inline docker-compose-base.yml]: `./aether-kernel:/code`

File: docker-compose-kernel.yml [hunk 13]
```diff
@@ -0,0 +1,57 @@
+# ------------------------------------------------------------------------------
+# Config file for only these containers:
+#
+#  * Aether Kernel Module
```
Comment (admbtlr) [inline docker-compose-kernel.yml]: `Aether Kernel`

File: docker-compose-odk.yml [hunk 14]
```diff
@@ -0,0 +1,76 @@
+# ------------------------------------------------------------------------------
+# Config file for only these containers:
+#
+#  * Aether Kernel Module
```
Comment (admbtlr) [inline docker-compose-odk.yml]: `Aether Kernel`

File: docker-compose-odk.yml [hunk 15]
```diff
@@ -0,0 +1,76 @@
...
+  db:
+    extends:
+      file: docker-compose-base.yml
+      service: postgres-base
+    networks:
+      - internal-odk
+
+
+  # ---------------------------------
+  # Aether Kernel module
```
Comment (admbtlr) [inline docker-compose-odk.yml]: `Aether Kernel`

File: docker-compose-sync.yml [hunk 16]
```diff
@@ -0,0 +1,100 @@
+# ------------------------------------------------------------------------------
+# Config file for only these containers:
+#
+#  * Aether Kernel Module
```
Comment (admbtlr) [inline docker-compose-sync.yml]: `Aether Kernel`

File: docker-compose-sync.yml [hunk 17]
```diff
@@ -0,0 +1,100 @@
...
+  redis:
+    extends:
+      file: docker-compose-base.yml
+      service: redis-base
+    networks:
+      - internal-sync
+
+
+  # ---------------------------------
+  # Aether Kernel module
```
Comment (admbtlr) [inline docker-compose-sync.yml]: `Aether Kernel`

File: docker-compose-sync.yml [hunk 18]
```diff
@@ -0,0 +1,100 @@
...
+    depends_on:
+      - db
+    networks:
+      internal-sync:
+        aliases:
+          - kernel.aether.local
+
+
+  # ---------------------------------
+  # Mobile Sync module
```
Comment (admbtlr) [inline docker-compose-sync.yml]: `CouchDB Sync Module`

File: docker-compose-test.yml [hunk 19]
```diff
@@ -1,150 +1,100 @@
+# ------------------------------------------------------------------------------
+# Config file for ALL TESTING containers:
+#
+#  * Aether Kernel Module
```
Comment (admbtlr) [inline docker-compose-test.yml]: `Aether Kernel`

File: docker-compose.yml [hunk 20]
```diff
@@ -1,217 +1,116 @@
+# ------------------------------------------------------------------------------
+# Config file for ALL containers:
+#
+#  * Aether Kernel Module
```
Comment (admbtlr) [inline docker-compose.yml]: `Aether Kernel`

File: docker-compose.yml [hunk 21]
```diff
@@ -1,217 +1,116 @@
...
+#  * ODK Collect Module
```
Comment (admbtlr) [inline docker-compose.yml]: `ODK Module`

File: docker-compose.yml [hunk 22]
```diff
@@ -1,217 +1,116 @@
...
+  redis:
+    extends:
+      file: docker-compose-base.yml
+      service: redis-base
+    networks:
+      - internal
+
+
+  # ---------------------------------
+  # Aether Kernel module
```
Comment (admbtlr) [inline docker-compose.yml]: `Aether Kernel`

**Review by admbtlr — CHANGES_REQUESTED** [review]
You must hate me by now :)

I think I've left a comment on every name change, so hopefully you can just go through them pretty quickly.

**Comment by admbtlr** [comment]
And to justify being a pain: developers who use Aether will be looking at and using these files; we need to get this small stuff exactly right so that we can have everyone who uses Aether speaking the same, unambiguous language.

**Review by quesebifurcan — COMMENTED** [review]
Looks good, great work! Unrelated to the quality of this PR, but I personally think having 7 different compose files with almost 700 lines of docker-compose configuration is a bit too much for a project this size. I also admit to not understanding exactly why we need the extra compose files. Would be great if @MrPink could take a closer look if he has time -- maybe there's a simpler way to set this up?

**Comment by obdulia-losantos** [comment]
@adamvert I hope this time is fine or close to! :sweat_smile:

**Review by admbtlr — APPROVED** [review]
🎉 💯

**Comment by obdulia-losantos** [comment]
:rocket: :roller_coaster:

**Comment by obdulia-losantos** [comment]
@MrPink whenever you have time! :grimacing: :pray:

**Review by MrPink — APPROVED** [review]

**Comment by obdulia-losantos** [comment]
@quesebifurcan talking to Will we decide to merge this and in another branch clean up the docker-compose files to make it simpler and clean. :see_no_evil:

**Pull request merged**